source/Octopus.Client/OctopusAsyncRepository.cs

```diff
@@ -176,6 +176,23 @@ public async Task<bool> HasLink(string name)
             var spaceRootDocument = await loadSpaceRootResource.Value.ConfigureAwait(false);
             return spaceRootDocument != null && spaceRootDocument.HasLink(name) || rootDocument.HasLink(name);
         }
+        
+        public async Task<bool> HasLinkParameter(string linkName, string parameterName)
+        {
+            string link;
+            var rootDocument = await loadRootResource.Value.ConfigureAwait(false);
```

Comment:
Is it worth moving this inside the `else` like in the synchronous version, so it only fetches when needed? I don't think it would be too verbose (although some braces might be warranted)